modin/pandas/test/dataframe/test_join_sort.py

```diff
@@ -527,6 +527,43 @@ def test_sort_values_with_string_index():
     df_equals(modin_df, pandas_df)
 
 
+@pytest.mark.skipif(
+    Engine.get() not in ["Ray", "Dask", "Unidist"],
+    reason="We only need to test this case where sort does not default to pandas.",
+)
+@pytest.mark.parametrize("ascending", [True, False], ids=["True", "False"])
+@pytest.mark.parametrize("na_position", ["first", "last"], ids=["first", "last"])
+def test_sort_values_with_only_one_non_na_row_in_partition(ascending, na_position):
+    pandas_df = pandas.DataFrame(
+        np.random.rand(1000, 100), columns=[f"col {i}" for i in range(100)]
+    )
+    # Need to ensure that one of the partitions has all NA values except for one row
+    pandas_df.iloc[340:] = np.NaN
+    pandas_df.iloc[-1] = -4.0
+    modin_df = pd.DataFrame(pandas_df)
+    modin_result = modin_df.sort_values(
+        "col 3", ascending=ascending, na_position=na_position
+    )
+    pandas_result = pandas_df.sort_values(
+        "col 3", ascending=ascending, na_position=na_position
+    )
+    df_equals(modin_result, pandas_result)
```

Comment:
```suggestion
    eval_general(modin_df, pandas_df, lamba df: df.sort_values("col 3", ascending=ascending, na_position=na_position))
```

 Probably need to reformat